# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5Highlighter.cpp
# Commit: 77388b0795b4
# Full Hash: 77388b0795b4f9a30171c922d6b649c52e5f6c87
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-23 09:34:36
# Description:
#   Bug 1745142 - Communicate encoding commitment via speculative load queue. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133996
# ==============================================================================

diff -r a388127095db -r 77388b0795b4 parser/html/nsHtml5Highlighter.cpp
--- a/parser/html/nsHtml5Highlighter.cpp	Wed Dec 22 15:47:08 2021 +0000
+++ b/parser/html/nsHtml5Highlighter.cpp	Wed Dec 22 15:54:49 2021 +0000
@@ -142,8 +142,11 @@
   // a speculation.
 
   mOpQueue.AppendElement()->Init(mozilla::AsVariant(opStartLayout()));
+}
 
-  FlushOps();
+void nsHtml5Highlighter::UpdateCharsetSource(nsCharsetSource aCharsetSource) {
+  opUpdateCharsetSource operation(aCharsetSource);
+  mOpQueue.AppendElement()->Init(mozilla::AsVariant(operation));
 }
 
 int32_t nsHtml5Highlighter::Transition(int32_t aState, bool aReconsume,